controllers/reconcilers/configuration/prometheus.go

```diff
@@ -473,6 +474,38 @@ func (r *Reconciler) reconcilePrometheus(ctx context.Context, cr *v1.Observabili
 	return nil
 }
 
+//resource.Quantity should be greater than than the default 250Gi
+func validateOverridePrometheusStorageSize(q resource.Quantity, cr *v1.Observability) bool {
+	return q.CmpInt64(250) >= 0
+}
+
+//construct Promnetheus storage spec wiht either default or override value from resources
+func getPrometheusStorageSpecHelper(cr *v1.Observability, indexes []v1.RepositoryIndex) (*prometheusv1.StorageSpec, error) {
+	prometheusStorageSpec := cr.Spec.Storage.PrometheusStorageSpec
+	customStorageSize := model.GetPrometheusStorageSize(cr, indexes)
+	q, err := resource.ParseQuantity(customStorageSize) //check if resources value is valid
+	if err != nil {
+		return prometheusStorageSpec, err
```

Comment:
Ok I see. This relies on the parse error (which also happens when the customStorageSize is empty). Better do an explicit check for an empty string here.

Reply:
There is a check on customStorageSize string on line [490](https://github.com/redhat-developer/observability-operator/pull/103/files/7e5bd538b5f868912a0afe8ef44d6f34bb6ae97d#diff-0160d4f43c9845ea36b44e45ee07136be67b03332226aae50e0ea3c0aed0f559R490).Should I check earlier?

Reply:
I think so, no need to parse it when it's empty.

Reply:
Good point!